lucille-core/src/test/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnectorTest.java

```diff
@@ -44,7 +43,48 @@ public void initTestMode() throws Exception {
   }
 
   @Test
-  public void testDatabaseConnector() throws Exception {
+  public void testDatabaseConnectorMixed() throws Exception {
+
+    // The only connection to the h2 database should be the dbHelper
+    assertEquals(1, dbHelper.checkNumConnections());
+
+    // Create the test config
+    HashMap<String, Object> configValues = new HashMap<>();
+    configValues.put("name", connectorName);
+    configValues.put("pipeline", pipelineName);
+    configValues.put("driver", "org.h2.Driver");
+    configValues.put("connectionString", "jdbc:h2:mem:test");
+    configValues.put("jdbcUser", "");
+    configValues.put("jdbcPassword", "");
+    configValues.put("sql", "select id,int_field,bool_field from mixed order by id");
+    configValues.put("idField", "id");
+
+    // create a config object off that map
+    Config config = ConfigFactory.parseMap(configValues);
+
+    // create the connector with the config
+    DatabaseConnector connector = new DatabaseConnector(config);
+
+    // start the connector
+    connector.execute(publisher);
+
+    // Confirm there were 3 results.
+    List<Document> docsSentForProcessing = messenger.getDocsSentForProcessing();
+    assertEquals(2, docsSentForProcessing.size());
+
+    assertEquals("1", docsSentForProcessing.get(0).getId());
+    assertEquals((Integer)3, docsSentForProcessing.get(0).getInt("int_field"));

```

Comment:
does this test break without the code change to DatabaseConnector (i.e. we'd like it to break but I'm wondering if it actually would). Calling getInt() on a JsonDocument field that had been set as a String should do an automatic conversions:

here's a test (not committed) that shows this:

```
  @Test
  public void testTypeConversion() {
    Document doc = createDocument("id1");
    doc.setField("field1", "1");
    assertEquals(Integer.valueOf(1), doc.getInt("field1"));
    assertEquals(String.class, doc.asMap().get("field1").getClass());
  }
```